pkg/antctl/raw/test/client.go

```diff
@@ -0,0 +1,182 @@
+	return c.clientSet.AppsV1().Deployments(namespace).Create(ctx, deployment, opts)
+}
+
+func (c *Client) GetDeployment(ctx context.Context, namespace, name string, opts metav1.GetOptions) (*appsv1.Deployment, error) {
+	return c.clientSet.AppsV1().Deployments(namespace).Get(ctx, name, opts)
+}
+
+func (c *Client) DeploymentIsReady(ctx context.Context, namespace, deploymentName string) (bool, error) {
+	deployment, err := c.GetDeployment(ctx, namespace, deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	if deployment.Generation <= deployment.Status.ObservedGeneration {
+		for _, cond := range deployment.Status.Conditions {
+			if cond.Type == appsv1.DeploymentProgressing && cond.Reason == "ProgressDeadlineExceeded" {
+				return false, fmt.Errorf("deployment %q exceeded its progress deadline", deployment.Name)
+			}
+		}
+		if deployment.Spec.Replicas != nil && deployment.Status.UpdatedReplicas < *deployment.Spec.Replicas {
+			return false, nil
+		}
+		if deployment.Status.Replicas > deployment.Status.UpdatedReplicas {
+			return false, nil
+		}
+		if deployment.Status.AvailableReplicas < deployment.Status.UpdatedReplicas {
+			return false, nil
+		}
+		return true, nil
+	}
+	return false, nil
+}
+
+func (c *Client) CreateNamespace(ctx context.Context, namespace string, opts metav1.CreateOptions) (*corev1.Namespace, error) {
+	return c.clientSet.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}, opts)
+}
+
+func (c *Client) GetNamespace(ctx context.Context, namespace string, options metav1.GetOptions) (*corev1.Namespace, error) {
+	return c.clientSet.CoreV1().Namespaces().Get(ctx, namespace, options)
+}
+
+func (c *Client) DeleteNamespace(ctx context.Context, namespace string, opts metav1.DeleteOptions) error {
+	return c.clientSet.CoreV1().Namespaces().Delete(ctx, namespace, opts)
+}
+
+func (c *Client) ListPods(ctx context.Context, namespace string, options metav1.ListOptions) (*corev1.PodList, error) {
+	return c.clientSet.CoreV1().Pods(namespace).List(ctx, options)
+}
+
+func (c *Client) ExecInPod(ctx context.Context, namespace, pod, container string, command []string) (bytes.Buffer, error) {
+	req := c.clientSet.CoreV1().RESTClient().Post().Resource("pods").Name(pod).Namespace(namespace).SubResource("exec")
+
+	scheme := runtime.NewScheme()
+	if err := corev1.AddToScheme(scheme); err != nil {
+		return bytes.Buffer{}, fmt.Errorf("error adding to scheme: %w", err)
+	}
+
+	parameterCodec := runtime.NewParameterCodec(scheme)
+
+	req.VersionedParams(&corev1.PodExecOptions{
+		Command:   command,
+		Container: container,
+		Stdin:     false,
+		Stdout:    true,
+		Stderr:    true,
+		TTY:       false,
+	}, parameterCodec)
+
+	exec, err := remotecommand.NewSPDYExecutor(c.config, "POST", req.URL())
+	if err != nil {
+		return bytes.Buffer{}, fmt.Errorf("error while creating executor: %w", err)
+	}
+
+	result := &ExecResult{}
```

Comment:
Given its usage, I think it's not useful to define this struct, you can just declare stdout and stderr, then use it in the function.

Reply:
Yes, Its better that way :)